ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -25,29 +25,47 @@ type SiteConfigBuilder struct {
 func NewSiteConfigBuilder() (*SiteConfigBuilder, error) {
 	scBuilder := SiteConfigBuilder{scBuilderExtraManifestPath: localExtraManifestPath}
 
-	clusterCRsYamls, err := scBuilder.splitYamls([]byte(clusterCRs))
+	return &scBuilder, nil
+}
+
+func (scbuilder *SiteConfigBuilder) loadSourceClusterCRs(clusterCRs string) error {
+	clusterCRsYamls, err := scbuilder.splitYamls([]byte(clusterCRs))
 	if err != nil {
-		return &scBuilder, err
+		return err
 	}
-	scBuilder.SourceClusterCRs = make([]interface{}, len(clusterCRsYamls))
+	scbuilder.SourceClusterCRs = make([]interface{}, len(clusterCRsYamls))
 	for id, clusterCRsYaml := range clusterCRsYamls {
 		var clusterCR interface{}
 		err := yaml.Unmarshal(clusterCRsYaml, &clusterCR)
 
 		if err != nil {
-			return &scBuilder, err
+			return err
 		}
-		scBuilder.SourceClusterCRs[id] = clusterCR
+		scbuilder.SourceClusterCRs[id] = clusterCR
 	}
 
-	return &scBuilder, nil
+	return nil
 }
 
 func (scbuilder *SiteConfigBuilder) SetLocalExtraManifestPath(path string) {
 	scbuilder.scBuilderExtraManifestPath = path
 }
 
 func (scbuilder *SiteConfigBuilder) Build(siteConfigTemp SiteConfig) (map[string][]interface{}, error) {
+	if siteConfigTemp.ApiVersion == siteConfigAPIV1 {
```

Comment:
if we extend API version later to v3,v4,v5, probably better to use switch instead of "else if" for better readability

Reply:
done